src/compute-client/src/controller.rs

```diff
@@ -165,6 +166,8 @@ pub struct ComputeController<T> {
     envd_epoch: NonZeroI64,
     /// The compute controller metrics.
     metrics: ComputeControllerMetrics,
+    /// A function that compute the lag between the given time and wallclock time.
+    wallclock_lag: WallclockLagFn<T>,
```

Comment:
Well, the controller certainly expects its caller to pass a function that uses the wallclock time to compute the lag! If the caller doesn't do that, it can't expect things to work properly. It would be nice to ensure that in the type system, but I don't think we can. We _could_ take a `NowFn` instead and compute the lag ourselves, but that would require us to make `T` even less generic then it already is.